src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -947,13 +948,30 @@ public void onProgressChanged(long numBytes,
 	}
 
 	/**
-	 * for Web Link and citation type of resources, append ".html" to the file name String
+	 * If file extension is missing, look up the extension by file MIME type
+	 * Then for Web Link and citation type of resources, append ".html" to the file name String
 	 * @param type
 	 * @param fileName
 	 * @return
 	 */
 	private String modifyFileNameOnType(String type, String fileName) {
-		if (Utils.CTOOLS_RESOURCE_TYPE_CITATION.equals(type) || Utils.CTOOLS_RESOURCE_TYPE_URL.equals(type))
+		if (!fileName.contains(Utils.FILE_EXTENSION_CHAR))
+		{
+			// do extension lookup first
+			try {
+				TikaConfig config = TikaConfig.getDefaultConfig();
+				MimeType mimeType = config.getMimeRepository().forName(type);
+				if (mimeType != null) {
+					String extension = mimeType.getExtension();
+				    //do something with the extension
+					fileName = fileName.concat(extension);
```

Comment:
Maybe it should be `fileName.concat(Utils.FILE_EXTENSION_CHAR).concat(extension)`.